client/post-editor/media-modal/detail/test/index.jsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { EditorMediaModalDetailItem as DetailItem } from '../detail-item';
+import { useSandbox } from 'test/helpers/use-sinon';
+
+jest.mock( 'post-editor/media-modal/detail/detail-fields', () =>
+	require( 'components/empty-component' )
+);
+jest.mock( 'post-editor/media-modal/detail/detail-file-info', () =>
+	require( 'components/empty-component' )
+);
+
+/**
+ * Module variables
+ */
+const DUMMY_SITE = { ID: 1 };
+
+const DUMMY_IMAGE_MEDIA = {
+	ID: 100,
+	date: '2015-06-19T11:36:09-04:00',
+	mime_type: 'image/jpeg',
+	guid: 'http://wordpress.com/test.jpg',
+	URL: 'http://wordpress.com/test.jpg',
+};
+
+const DUMMY_VIDEO_MEDIA = {
+	ID: 200,
+	date: '2015-06-19T11:36:09-04:00',
+	mime_type: 'video/mp4',
+	videopress_guid: 'testvideopressguid123',
+	guid: 'http://wordpress.com/test.mp4',
+	URL: 'http://wordpress.com/test.mp4',
+};
+
+const SHARED_PROPS = {
+	site: DUMMY_SITE,
+	canUserUploadFiles: true,
+	translate: str => str,
+};
+
+describe( 'EditorMediaModalDetailItem', () => {
+	let isVideoPressEnabled;
+
+	useSandbox( sandbox => {
+		isVideoPressEnabled = sandbox.stub().returns( true );
+	} );
+
+	test( 'should display edit button for a VideoPress video on a public site', () => {
+		const tree = shallow(
+			<DetailItem
+				item={ DUMMY_VIDEO_MEDIA }
+				isVideoPressEnabled={ isVideoPressEnabled }
+				{ ...SHARED_PROPS }
+			/>
+		);
+
+		const editButton = tree.find( '.editor-media-modal-detail__edit' );
+
+		// 2 Edit buttons are rendered. It's recommended te have exact value tests.
+		// Not sure whethur to change this to
+		// expect( editButton ).to.have.length(2);
```

Comment:
+1 for being more explicit with the count here...

The trouble is, it doesn't make sense that the validation of `should display edit button for a VideoPress video on a public site` is to see exactly 2 `editButtons`.

It might be worth being more specific with the edit buttons - maybe we're concerned with two test cases that each check one of the 2 expected edit buttons.

I'm not totally against `.to.have.length.at.least( 1 );` though, if that is truly what we're concerned with, if all we're interested in is that somewhere there is an edit button and we don't care too much about any additional edit buttons. If that's the case I'd then consider weaving this in to the test description: `should display at least one edit button ...`

Reply:
It's more that we want to make sure there's at least one edit button. I've updated the test descriptions.

Reply:
Brilliant, that makes sense to me :) Thanks for updating that!